src/workerd/api/sockets.h

```diff
@@ -55,10 +69,11 @@ class Socket: public jsg::Object {
   kj::Own<jsg::PromiseResolverPair<void>> closeFulfiller;
   // This fulfiller is used to resolve the `closedPromise` below.
   jsg::MemoizedIdentity<jsg::Promise<void>> closedPromise;
+  kj::Promise<void> writeDisconnectedPromise;
+  jsg::Optional<SocketOptions> options;
 
   kj::Promise<kj::Own<kj::AsyncIoStream>> processConnection();
-
-
+  jsg::Promise<void> closeWriteSide(jsg::Lock& js);
```

Comment:
Given that this might not actually close the writable site, maybe the name `maybeCloseWriteSide` would be appropriate?